### PR TITLE
perf improvements and klusterlet customizations

### PR DESCRIPTION
**Pull request by xrajesh**
Signed-off-by: Xavier Dharmaiyan <xavier@redhat.com>

**Related Issue:**  stolostron/backlog#18606 and stolostron/backlog#25847

### Description of changes
- Adds annotations to support override of default configuration in add-on deployment.
- Reduce scope of resource watches to reduce memory usage.

### CODE REVIEW

File: addon/manifests/chart/templates/deployment.yaml [hunk 1]
```diff
@@ -24,6 +24,10 @@ spec:
       - name: collector
         image: "{{ .Values.global.imageOverrides.search_collector }}"
         imagePullPolicy: {{ .Values.global.imagePullPolicy }}
+                {{- if .Values.userargs.containerArgs }}
```
Comment (jlpadilla) [inline addon/manifests/chart/templates/deployment.yaml:27]: indentation might be wrong.

Comment (xrajesh) [reply addon/manifests/chart/templates/deployment.yaml]: Fixed it

File: controllers/search_controller.go [hunk 2]
```diff
@@ -205,12 +203,14 @@ func (r *SearchReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
-		Watches(&source.Kind{Type: &rbacv1.ClusterRoleBinding{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(pred)).
-		Watches(&source.Kind{Type: &rbacv1.ClusterRole{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(pred)).
+		Watches(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &searchv1alpha1.Search{},
+		}, builder.WithPredicates(pred)).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &searchv1alpha1.Search{},
+		}, builder.WithPredicates(pred)).
```
Comment (jlpadilla) [inline controllers/search_controller.go:213]: Curious why we aren't watching other resources anymore.
Would the result be that we could get to a broken state if any of the resources not watched change?

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

File: controllers/search_controller.go (hunk 2 above)
Comment (xrajesh) [reply controllers/search_controller.go:213]: Reducing the reconciliation only to Deployment changes and secrets .

**Pull request merged**